www/guides/react.md

```diff
@@ -0,0 +1,361 @@
+
+- How to set up Snowpack
+- How to get started with React in a Snowpack project
+- Importing CSS, images and other assets
+- Enabling [Fast Refresh](https://reactnative.dev/docs/fast-refresh) mode for React
+- Connecting your favorite developer tools to Snowpack
+
+> 💡 Tip: if you want to jump to the end to see a full featured React setup, the [Create Snowpack App React template](https://github.com/pikapkg/snowpack/tree/master/create-snowpack-app/app-template-react) comes with everything you'll learn in this tutorial plus other useful tools.
+>
+> ```bash
+> npx create-snowpack-project react-snowpack-template --template @snowpack/project-template-react
+> ```
+
+Prerequisites: Snowpack is a command line tool installed from npm. This guide assumes a basic understanding of Node.js, npm, and how to run commands in the terminal. Knowledge of React is not required, Snowpack is a great way to learn React!
+
+## Getting started with Snowpack and React
+
+The easiest way to start a new Snowpack project is with [Create Snowpack App](<https://www.snowpack.dev/#create-snowpack-app-(csa)>). Create Snowpack App is a tool to quickly set up Snowpack in a new directory.
+
+The template you're starting with here is `@snowpack/project-template-minimal` is a bare-bones setup where Snowpack serves an `index.html`, `index.css`, and `index.js`.
+
+Open up your terminal and head to a directory where you want to put your new project. Now run `npx create-snowpack-project react-snowpack --template @snowpack/project-template-minimal`. This creates a new directory called `react-snowpack` and installs the `@snowpack/project-template-minimal` inside it.
+
+You can now head to that new `react-snowpack` directory with `cd react-snowpack`. And to start up Snowpack run `npm run start`. You should see your new website up and running!
+
+> 💡 Tip: the `README.md` in your new project contains useful information about what each file does.
+
+Image: screenshot of project-template-minimal, which shows "Hello world" in text on a white background.
+
+> 💡 Tip: you can also use Yarn for any of the commands in this tutorial.
+
+Now that you have a basic project up and running, it's time to install React. To install React, run the following command in your project directory:
+
+```bash
+npm install react react-dom --save
+```
+
+> 💡 Tip: in this tutorial commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+Snowpack has built in support for JSX in files with the `.jsx` extension. That means that there's no additional plugins or configuration needed to write your first React component, but you will need to rename your `index.js` file to `index.jsx` to take advantage of this.
```

Comment:
New user: "what's JSX?"

Prefix this paragraph with "React relies on a special templating language called JSX. Snowpack has..."